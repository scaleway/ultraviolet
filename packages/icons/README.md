# Ultraviolet Icons

[![npm version](https://badge.fury.io/js/%40ultraviolet%2Ficons.svg)](https://badge.fury.io/js/%40ultraviolet%2Ficons)

Ultraviolet Icons is a set of components based on SVG icons. It provides a set of icons that can be used in your projects.

## Get Started

```sh
$ pnpm add @ultraviolet/icons @ultraviolet/themes
```

You can then add `ThemeProvider` to your applications and use the provided theme from `@ultraviolet/themes` or use your own.

```tsx
import { ThemeProvider } from '@emotion/react'
import { SdkGoProductIcon } from '@ultraviolet/icons/product'
import { theme } from '@ultraviolet/themes'

const App = () => (
  <ThemeProvider theme={theme}>
    <SdkGoProductIcon size="medium" variant="primary"/>
  </ThemeProvider>
)
```

> **Note**:
> To generate your own theme easily you can check the [theme generator](https://storybook.ultraviolet.scaleway.com/?path=/docs/tools-theme-generator--docs).

N.B. To allow typescript theme typings with `@emotion/styled` components,
you'll have to define the `@emotion/react` module `Theme` interface in your project.

Example, in a `emotion.d.ts` file:

```ts
import '@emotion/react'
import type { consoleLightTheme } from '@ultraviolet/themes'

type UltravioletUITheme = typeof consoleLightTheme

declare module '@emotion/react' {
  export interface Theme extends UltravioletUITheme {}
}
```

## Documentation

Checkout our [documentation website](https://storybook.ultraviolet.scaleway.com/).

## Contributing

üìù You can participate in the development and [start contributing](/CONTRIBUTING.md) to it.
