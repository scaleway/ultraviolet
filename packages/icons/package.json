{
  "name": "@ultraviolet/icons",
  "version": "4.0.0-beta.1",
  "description": "Ultraviolet Icons",
  "homepage": "https://github.com/scaleway/ultraviolet#readme",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/scaleway/ultraviolet.git",
    "directory": "packages/icons"
  },
  "scripts": {
    "type:generate": "tsc --declaration -p tsconfig.build.json",
    "watch": "pnpm run '/^watch:.*/'",
    "watch:build": "vite build --config vite.config.ts --watch",
    "build": "vite build --config vite.config.ts && pnpm run type:generate",
    "build:profile": "npx vite-bundle-visualizer -c vite.config.ts",
    "typecheck": "tsc --noEmit",
    "size": "size-limit",
    "lintpublish": "publint"
  },
  "keywords": [
    "react",
    "reactjs",
    "ui"
  ],
  "license": "Apache-2.0",
  "publishConfig": {
    "access": "public"
  },
  "engines": {
    "node": ">=18.x",
    "pnpm": ">=9.x"
  },
  "os": [
    "darwin",
    "linux"
  ],
  "sideEffects": false,
  "type": "module",
  "module": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "main": "./dist/index.cjs",
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "require": "./dist/index.cjs",
      "import": "./dist/index.js",
      "default": "./dist/index.js"
    },
    "./*": {
      "types": "./dist/components/Icon/__generated__/*.d.ts",
      "require": "./dist/components/Icon/__generated__/*.cjs",
      "import": "./dist/components/Icon/__generated__/*.js",
      "default": "./dist/components/Icon/__generated__/*.js"
    },
    "./product": {
      "types": "./dist/components/ProductIcon/index.d.ts",
      "require": "./dist/components/ProductIcon/index.cjs",
      "import": "./dist/components/ProductIcon/index.js",
      "default": "./dist/components/ProductIcon/index.js"
    },
    "./product/*": {
      "types": "./dist/components/ProductIcon/__generated__/*.d.ts",
      "require": "./dist/components/ProductIcon/__generated__/*.cjs",
      "import": "./dist/components/ProductIcon/__generated__/*.js",
      "default": "./dist/components/ProductIcon/__generated__/*.js"
    },
    "./category": {
      "types": "./dist/components/CategoryIcon/index.d.ts",
      "require": "./dist/components/CategoryIcon/index.cjs",
      "import": "./dist/components/CategoryIcon/index.js",
      "default": "./dist/components/CategoryIcon/index.js"
    },
    "./category/*": {
      "types": "./dist/components/CategoryIcon/__generated__/*.d.ts",
      "require": "./dist/components/CategoryIcon/__generated__/*.cjs",
      "import": "./dist/components/CategoryIcon/__generated__/*.js",
      "default": "./dist/components/CategoryIcon/__generated__/*.js"
    },
    "./logo": {
      "types": "./dist/components/Logo/index.d.ts",
      "require": "./dist/components/Logo/index.cjs",
      "import": "./dist/components/Logo/index.js",
      "default": "./dist/components/Logo/index.js"
    },
    "./logo/*": {
      "types": "./dist/components/Logo/__generated__/*.d.ts",
      "require": "./dist/components/Logo/__generated__/*.cjs",
      "import": "./dist/components/Logo/__generated__/*.js",
      "default": "./dist/components/Logo/__generated__/*.js"
    }
  },
  "size-limit": [
    {
      "path": [
        "dist/**/*.js",
        "!*.d.ts",
        "!*.cjs"
      ],
      "limit": "600 kB",
      "webpack": false,
      "brotli": true,
      "running": false
    }
  ],
  "peerDependencies": {
    "@emotion/react": "11.14.0",
    "@emotion/styled": "11.14.0",
    "react": "18.x || 19.x",
    "react-dom": "18.x || 19.x"
  },
  "devDependencies": {
    "@babel/core": "7.26.10",
    "@emotion/react": "11.14.0",
    "@emotion/styled": "11.14.0",
    "@types/react": "19.0.12",
    "@types/react-dom": "19.0.4",
    "@utils/test": "workspace:*",
    "react": "19.0.0",
    "react-dom": "19.0.0"
  },
  "dependencies": {
    "@ultraviolet/themes": "workspace:*"
  }
}
