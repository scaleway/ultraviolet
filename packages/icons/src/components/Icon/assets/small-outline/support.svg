<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M5.70123 2.79591L7.34319 4.76626C7.77639 4.67861 8.22361 4.67861 8.65681 4.76626L10.2988 2.79591C8.8385 2.15136 7.1615 2.15136 5.70123 2.79591ZM11.4435 3.47245L9.91187 5.31038C10.0597 5.4157 10.2007 5.53457 10.3331 5.66692C10.4654 5.79926 10.5843 5.94026 10.6896 6.08813L12.5275 4.55653C12.3743 4.35558 12.2057 4.16233 12.0217 3.97833C11.8377 3.79433 11.6444 3.62573 11.4435 3.47245ZM13.2041 5.70123L11.2337 7.34319C11.3214 7.77639 11.3214 8.22361 11.2337 8.65681L13.2041 10.2988C13.8486 8.8385 13.8486 7.1615 13.2041 5.70123ZM12.5275 11.4435L10.6896 9.91187C10.5843 10.0597 10.4654 10.2007 10.3331 10.3331C10.2007 10.4654 10.0597 10.5843 9.91187 10.6896L11.4435 12.5275C11.6444 12.3743 11.8377 12.2057 12.0217 12.0217C12.2057 11.8377 12.3743 11.6444 12.5275 11.4435ZM10.2988 13.2041L8.65681 11.2337C8.22361 11.3214 7.77639 11.3214 7.34319 11.2337L5.70123 13.2041C7.1615 13.8486 8.8385 13.8486 10.2988 13.2041ZM4.55653 12.5275L6.08813 10.6896C5.94026 10.5843 5.79926 10.4654 5.66692 10.3331C5.53457 10.2007 5.4157 10.0597 5.31038 9.91187L3.47245 11.4435C3.62573 11.6444 3.79433 11.8377 3.97833 12.0217C4.16233 12.2057 4.35558 12.3743 4.55653 12.5275ZM2.79591 10.2988L4.76626 8.65681C4.67861 8.22361 4.67861 7.77639 4.76626 7.34319L2.79591 5.70123C2.15136 7.1615 2.15136 8.8385 2.79591 10.2988ZM3.47245 4.55653L5.31038 6.08813C5.4157 5.94026 5.53457 5.79926 5.66692 5.66692C5.79926 5.53457 5.94026 5.4157 6.08813 5.31038L4.55653 3.47245C4.35558 3.62573 4.16233 3.79433 3.97833 3.97833C3.79433 4.16233 3.62573 4.35558 3.47245 4.55653ZM4.33443 2.03478C6.5759 0.655073 9.4241 0.655073 11.6656 2.03478C12.1229 2.3163 12.5544 2.65494 12.9497 3.05025C13.3451 3.44557 13.6837 3.87707 13.9652 4.33443C15.3449 6.5759 15.3449 9.4241 13.9652 11.6656C13.6837 12.1229 13.3451 12.5544 12.9497 12.9497C12.5544 13.3451 12.1229 13.6837 11.6656 13.9652C9.4241 15.3449 6.5759 15.3449 4.33443 13.9652C3.87707 13.6837 3.44556 13.3451 3.05025 12.9497C2.65494 12.5544 2.3163 12.1229 2.03478 11.6656C0.655073 9.4241 0.655073 6.5759 2.03478 4.33443C2.3163 3.87707 2.65494 3.44557 3.05025 3.05025C3.44557 2.65494 3.87707 2.3163 4.33443 2.03478ZM8.67003 6.12874C8.23788 5.97445 7.76212 5.97445 7.32998 6.12874C7.06241 6.22427 6.81082 6.37917 6.59499 6.59499C6.37917 6.81082 6.22427 7.06241 6.12874 7.32998C5.97445 7.76212 5.97445 8.23789 6.12874 8.67003C6.22427 8.93759 6.37917 9.18918 6.59499 9.40501C6.81082 9.62084 7.06241 9.77573 7.32997 9.87126C7.76212 10.0256 8.23788 10.0256 8.67003 9.87126C8.93759 9.77573 9.18918 9.62084 9.40501 9.40501C9.62084 9.18918 9.77573 8.93759 9.87126 8.67003C10.0256 8.23789 10.0256 7.76212 9.87126 7.32998C9.77573 7.06241 9.62084 6.81082 9.40501 6.59499C9.18918 6.37917 8.93759 6.22427 8.67003 6.12874Z" fill="black"/>
</svg>
