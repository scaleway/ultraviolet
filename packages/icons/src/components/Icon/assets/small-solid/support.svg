<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="none" viewBox="0 0 16 16">
    <path d="M12.95 3.05a7 7 0 1 0-9.9 9.9 7 7 0 0 0 9.9-9.9m-7.262-.042a5.52 5.52 0 0 1 4.624 0L8.698 5.082a3 3 0 0 0-1.396 0zm-2.68 2.68a5.52 5.52 0 0 0 0 4.624l2.074-1.614a3 3 0 0 1 0-1.396zm2.68 7.304 1.614-2.074c.458.11.938.11 1.396 0l1.614 2.074a5.52 5.52 0 0 1-4.624 0m7.304-2.68a5.52 5.52 0 0 0 0-4.624l-2.074 1.614c.11.458.11.938 0 1.396zM6.94 6.939a1.5 1.5 0 1 1 2.122 2.122A1.5 1.5 0 0 1 6.94 6.939" clip-rule="evenodd"/>
</svg>
