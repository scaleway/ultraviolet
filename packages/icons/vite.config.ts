import { resolve } from 'path'
import svgr from '@svgr/rollup'
import { defineConfig, mergeConfig } from 'vite'
import { defaultConfig } from '../../vite.config'

export default mergeConfig(defineConfig(defaultConfig), {
  plugins: [
    svgr({ memo: true, svgo: false }), // We disable svgo because we have custom configuration for it svgo.config.js
  ],
  build: {
    lib: {
      entry: {
        '': resolve(__dirname, 'src/index.ts'),
        'components/ProductIcon/index': resolve(
          __dirname,
          'src/components/ProductIcon/index.ts',
        ),
        'components/CategoryIcon/index': resolve(
          __dirname,
          'src/components/CategoryIcon/index.ts',
        ),
        'components/Logo/index': resolve(
          __dirname,
          'src/components/Logo/index.ts',
        ),
      },
    },
  },
})
