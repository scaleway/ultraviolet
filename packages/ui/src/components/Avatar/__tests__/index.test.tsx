import { screen } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'
import { MosaicIcon } from '@ultraviolet/icons'
import { renderWithTheme, shouldMatchEmotionSnapshot } from '@utils/test'
import { describe, expect, it, vi } from 'vitest'
import { Avatar } from '..'
import support from '../__stories__/assets/avatar.svg'

describe('Avatar', () => {
  describe.each(['circle', 'square'] as const)(
    `renders correctly with shape %s`,
    shape => {
      describe.each(['xsmall', 'small', 'medium', 'large'] as const)(
        `renders correctly with shape ${shape} and size %s`,
        size => {
          it('renders correctly with variant user', () =>
            shouldMatchEmotionSnapshot(
              <Avatar shape={shape} size={size} variant="user" />,
            ))

          it('renders correctly with variant image', () =>
            shouldMatchEmotionSnapshot(
              <Avatar
                image={support}
                shape={shape}
                size={size}
                variant="image"
              />,
            ))

          it('renders correctly with variant text', () =>
            shouldMatchEmotionSnapshot(
              <Avatar shape={shape} size={size} text="UV" variant="text" />,
            ))

          it('renders correctly with variant text and sentiment neutral', () =>
            shouldMatchEmotionSnapshot(
              <Avatar
                sentiment="neutral"
                shape={shape}
                size={size}
                text="UV"
                variant="text"
              />,
            ))

          it('renders correctly with variant icon', () =>
            shouldMatchEmotionSnapshot(
              <Avatar shape={shape} size={size} variant="icon">
                <MosaicIcon />
              </Avatar>,
            ))

          it('renders correctly with variant icon and sentiment neutral', () =>
            shouldMatchEmotionSnapshot(
              <Avatar
                sentiment="neutral"
                shape={shape}
                size={size}
                variant="icon"
              >
                <MosaicIcon />
              </Avatar>,
            ))

          it('renders correctly with variant colors', () =>
            shouldMatchEmotionSnapshot(
              <Avatar shape={shape} size={size} variant="colors" />,
            ))

          it('renders correctly with variant text and upload', async () => {
            const onClick = vi.fn()
            const { asFragment } = renderWithTheme(
              <Avatar
                data-testid="avatar"
                onClick={onClick}
                shape={shape}
                size={size}
                text="UV"
                upload
                variant="text"
              />,
            )
            const avatar = screen.getByTestId('avatar')

            await userEvent.hover(avatar)
            expect(asFragment()).toMatchSnapshot()

            await userEvent.click(avatar)
            expect(onClick).toHaveBeenCalledOnce()
          })
        },
      )
    },
  )
})
