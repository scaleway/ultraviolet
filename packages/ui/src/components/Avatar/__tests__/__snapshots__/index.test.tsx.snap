// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar > renders correctly with custom background color 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 32px;
  width: 32px;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFFFFF;
  border-radius: 100%;
  color: #ffffff;
  font-size: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
      size="32"
    >
      <span
        class="emotion-2 emotion-3"
      >
        HW
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Avatar > renders correctly with custom text color 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 32px;
  width: 32px;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #b824f9;
  border-radius: 100%;
  color: #FFFFFF;
  font-size: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
      size="32"
    >
      <span
        class="emotion-2 emotion-3"
      >
        HW
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Avatar > renders correctly with default props 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 32px;
  width: 32px;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
      size="32"
    >
      <img
        alt=""
        height="100%"
        src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20viewBox='0%200%2032%2032'%3e%3cg%20fill='none'%20fill-rule='evenodd'%3e%3cpath%20fill='%23EEF'%20d='m18.592.849%2011.59%208.488a4.46%204.46%200%200%201%201.602%204.969L27.357%2028.04a4.41%204.41%200%200%201-4.194%203.071H8.837a4.41%204.41%200%200%201-4.194-3.07L.216%2014.305a4.46%204.46%200%200%201%201.602-4.97L13.408.85a4.38%204.38%200%200%201%205.184%200z'/%3e%3cg%20fill-rule='nonzero'%3e%3cpath%20fill='%234F0599'%20d='M7.345%2019.765a9.1%209.1%200%200%200%205.87%205.883l-.154%201a10.1%2010.1%200%200%201-6.717-6.734zm17.293.053.998.161a10.1%2010.1%200%200%201-6.697%206.67l-.154-1a9.1%209.1%200%200%200%205.853-5.83zM13.04%207.358l.158%201a9.1%209.1%200%200%200-5.853%205.877l-1-.15a10.1%2010.1%200%200%201%206.695-6.727m5.92%200a10.1%2010.1%200%200%201%206.676%206.663l-.998.16a9.1%209.1%200%200%200-5.836-5.824z'/%3e%3cpath%20fill='%23AE5CFF'%20d='m13.422%2021.007.988.152-.783%205.097.067.01a17%2017%200%200%200%201.718.141l.588.01c.788%200%201.561-.051%202.306-.15l.066-.011-.783-5.097.989-.151.86%205.592a.5.5%200%200%201-.412.569%2018%2018%200%200%201-3.026.248%2018%2018%200%200%201-3.027-.248.5.5%200%200%201-.416-.48l.006-.09zm12.747-7.075c.164.981.248%202.012.248%203.068s-.084%202.087-.248%203.068a.5.5%200%200%201-.483.418l-.09-.007-5.592-.902.16-.987%205.091.821.011-.07c.075-.566.122-1.15.141-1.743l.01-.598c0-.8-.051-1.586-.15-2.342l-.012-.07-5.092.822-.16-.988%205.593-.902a.5.5%200%200%201%20.573.412m-19.86-.349.089.005%205.592.834-.147.989-5.1-.761-.01.065a17%2017%200%200%200-.14%201.703l-.01.582c0%20.78.051%201.546.15%202.285l.01.063%205.1-.759.147.989-5.592.834a.5.5%200%200%201-.567-.41A18%2018%200%200%201%205.583%2017c0-1.032.084-2.04.248-3.001a.5.5%200%200%201%20.478-.416m9.691-7c1.049%200%202.073.084%203.048.248a.5.5%200%200%201%20.417.482l-.006.09-.882%205.591-.987-.155.802-5.096-.068-.01a18%2018%200%200%200-1.731-.14L16%207.583c-.795%200-1.574.051-2.324.15l-.07.01.804%205.096-.988.156-.88-5.593a.5.5%200%200%201%20.41-.57c.975-.165%202-.249%203.048-.249'/%3e%3cpath%20fill='%234F0599'%20d='M16%2011.917a5.083%205.083%200%201%200%200%2010.166%205.083%205.083%200%200%200%200-10.166m0%201a4.083%204.083%200%201%201%200%208.166%204.083%204.083%200%200%201%200-8.166'/%3e%3c/g%3e%3c/g%3e%3c/svg%3e"
        width="100%"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Avatar > renders correctly with lock 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 32px;
  width: 32px;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #e9eaeb;
  border-radius: 100%;
  color: #ffffff;
  font-size: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
}

.emotion-4 {
  vertical-align: middle;
  fill: #727683;
  height: 1rem;
  width: 1rem;
  min-width: 1rem;
  min-height: 1rem;
}

.emotion-4 .fillStroke {
  stroke: #727683;
  fill: none;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
      size="32"
    >
      <span
        class="emotion-2 emotion-3"
      >
        <svg
          class="emotion-4 emotion-5"
          viewBox="0 0 16 16"
        >
          <path
            clip-rule="evenodd"
            d="M8 1C6.067 1 4.5 2.567 4.5 4.5L4.5 7C3.67157 7 3 7.67157 3 8.5V13.5C3 14.3284 3.67157 15 4.5 15H11.5C12.3284 15 13 14.3284 13 13.5V8.5C13 7.67157 12.3284 7 11.5 7V4.5C11.5 2.567 9.933 1 8 1ZM10 7V4.5C10 3.39543 9.10457 2.5 8 2.5C6.89543 2.5 6 3.39543 6 4.5V7H10Z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Avatar > renders correctly with sentence 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 32px;
  width: 32px;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #b824f9;
  border-radius: 100%;
  color: #ffffff;
  font-size: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
      size="32"
    >
      <span
        class="emotion-2 emotion-3"
      >
        HW
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Avatar > renders correctly with size 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 48px;
  width: 48px;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
      size="48"
    >
      <img
        alt=""
        height="100%"
        src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20viewBox='0%200%2032%2032'%3e%3cg%20fill='none'%20fill-rule='evenodd'%3e%3cpath%20fill='%23EEF'%20d='m18.592.849%2011.59%208.488a4.46%204.46%200%200%201%201.602%204.969L27.357%2028.04a4.41%204.41%200%200%201-4.194%203.071H8.837a4.41%204.41%200%200%201-4.194-3.07L.216%2014.305a4.46%204.46%200%200%201%201.602-4.97L13.408.85a4.38%204.38%200%200%201%205.184%200z'/%3e%3cg%20fill-rule='nonzero'%3e%3cpath%20fill='%234F0599'%20d='M7.345%2019.765a9.1%209.1%200%200%200%205.87%205.883l-.154%201a10.1%2010.1%200%200%201-6.717-6.734zm17.293.053.998.161a10.1%2010.1%200%200%201-6.697%206.67l-.154-1a9.1%209.1%200%200%200%205.853-5.83zM13.04%207.358l.158%201a9.1%209.1%200%200%200-5.853%205.877l-1-.15a10.1%2010.1%200%200%201%206.695-6.727m5.92%200a10.1%2010.1%200%200%201%206.676%206.663l-.998.16a9.1%209.1%200%200%200-5.836-5.824z'/%3e%3cpath%20fill='%23AE5CFF'%20d='m13.422%2021.007.988.152-.783%205.097.067.01a17%2017%200%200%200%201.718.141l.588.01c.788%200%201.561-.051%202.306-.15l.066-.011-.783-5.097.989-.151.86%205.592a.5.5%200%200%201-.412.569%2018%2018%200%200%201-3.026.248%2018%2018%200%200%201-3.027-.248.5.5%200%200%201-.416-.48l.006-.09zm12.747-7.075c.164.981.248%202.012.248%203.068s-.084%202.087-.248%203.068a.5.5%200%200%201-.483.418l-.09-.007-5.592-.902.16-.987%205.091.821.011-.07c.075-.566.122-1.15.141-1.743l.01-.598c0-.8-.051-1.586-.15-2.342l-.012-.07-5.092.822-.16-.988%205.593-.902a.5.5%200%200%201%20.573.412m-19.86-.349.089.005%205.592.834-.147.989-5.1-.761-.01.065a17%2017%200%200%200-.14%201.703l-.01.582c0%20.78.051%201.546.15%202.285l.01.063%205.1-.759.147.989-5.592.834a.5.5%200%200%201-.567-.41A18%2018%200%200%201%205.583%2017c0-1.032.084-2.04.248-3.001a.5.5%200%200%201%20.478-.416m9.691-7c1.049%200%202.073.084%203.048.248a.5.5%200%200%201%20.417.482l-.006.09-.882%205.591-.987-.155.802-5.096-.068-.01a18%2018%200%200%200-1.731-.14L16%207.583c-.795%200-1.574.051-2.324.15l-.07.01.804%205.096-.988.156-.88-5.593a.5.5%200%200%201%20.41-.57c.975-.165%202-.249%203.048-.249'/%3e%3cpath%20fill='%234F0599'%20d='M16%2011.917a5.083%205.083%200%201%200%200%2010.166%205.083%205.083%200%200%200%200-10.166m0%201a4.083%204.083%200%201%201%200%208.166%204.083%204.083%200%200%201%200-8.166'/%3e%3c/g%3e%3c/g%3e%3c/svg%3e"
        width="100%"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Avatar > renders correctly with small text 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 32px;
  width: 32px;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #b824f9;
  border-radius: 100%;
  color: #ffffff;
  font-size: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
      size="32"
    >
      <span
        class="emotion-2 emotion-3"
      >
        HW
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Avatar > renders correctly with text 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 32px;
  width: 32px;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #b824f9;
  border-radius: 100%;
  color: #ffffff;
  font-size: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
      size="32"
    >
      <span
        class="emotion-2 emotion-3"
      >
        HE
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Avatar > renders correctly with text and custom style 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 32px;
  width: 32px;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: red;
  border-radius: 100%;
  color: white;
  font-size: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
      size="32"
    >
      <span
        class="emotion-2 emotion-3"
      >
        HW
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Avatar > renders correctly with text size 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 50px;
  width: 50px;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #b824f9;
  border-radius: 100%;
  color: #ffffff;
  font-size: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
}

<div
    data-testid="testing"
  >
    <div
      class="emotion-0 emotion-1"
      size="50"
    >
      <span
        class="emotion-2 emotion-3"
      >
        HE
      </span>
    </div>
  </div>
</DocumentFragment>
`;
